src/__tests__/index.test.ts

```diff
@@ -1,23 +1,43 @@
-import delay from 'delay';
-import LRUCache from 'lru-cache';
-import axios, { AxiosInstance } from 'axios';
-import * as axiosCachingDns from '..';
+import axios, { AxiosInstance } from "axios";
+import delay from "delay";
+import * as axiosCachingDns from "..";
+import { DNSEntryCache } from "../DNSEntryCache";
 
 let axiosClient: AxiosInstance;
+let useRedis: boolean = false; // Flag to switch between Redis and LRU
 
-beforeEach(() => {
+beforeEach(async () => {
   axiosCachingDns.config.dnsTtlMs = 1000;
   axiosCachingDns.config.dnsIdleTtlMs = 5000;
   axiosCachingDns.config.cacheGraceExpireMultiplier = 2;
   axiosCachingDns.config.backgroundScanMs = 100;
 
-  axiosCachingDns.cacheConfig.ttl = (axiosCachingDns.config.dnsTtlMs * axiosCachingDns.config.cacheGraceExpireMultiplier);
+  if (useRedis) {
+    axiosCachingDns.config.redisConfig = {
+      url: "localhost:6379",
+      ttl:
+        (axiosCachingDns.config.dnsTtlMs *
+          axiosCachingDns.config.cacheGraceExpireMultiplier) /
+        1000,
+    };
+    axiosCachingDns.config.lruCacheConfig = undefined;
+  } else {
+    axiosCachingDns.config.lruCacheConfig = {
+      max: axiosCachingDns.config.dnsCacheSize,
+      ttl:
+        axiosCachingDns.config.dnsTtlMs *
+        axiosCachingDns.config.cacheGraceExpireMultiplier, // grace for refresh
+    };
+    axiosCachingDns.config.redisConfig = undefined;
+  }
 
-  axiosCachingDns.config.cache = new LRUCache(axiosCachingDns.cacheConfig);
+  axiosCachingDns.config.cache = new DNSEntryCache(
+    axiosCachingDns.config,
+    true // Create a fresh cache on each run
+  );
```

Comment:
We create a fresh cache on each run since redis is(/can be) a persistent cache and we dont want previous test runs' stored entries to affect current runs

We're also using the `DNSEntryCache` wrapper here so that we handle creating a redis or lru-cache depending on what the user has configured